src/dotnet-interactive-vscode-ads/src/versionSpecificFunctions.ts

```diff
@@ -27,3 +28,19 @@ export async function replaceNotebookMetadata(notebookUri: vscode.Uri, documentM
     const succeeded = await vscode.workspace.applyEdit(edit);
     return succeeded;
 }
+
+export async function handleRequestInput(prompt: string, password: boolean, inputTypeHint: string): Promise<string | null | undefined> {
+    let result: string | null | undefined;
+    if (inputTypeHint === 'connectionstring-mssql') {
+        let connection = await azdata.connection.openConnectionDialog();
+        if (connection) {
+            result = await azdata.connection.getConnectionString(connection.connectionId, true);
+        }
+    } else {
+        result = (inputTypeHint === 'file')
```

Comment:
It would be ideal if this block wasn't repeated code, e.g. a map of type hint to implementation that the ADS code could just add a handler to. The code is simple now but additional use of type hints is planned and it will get more complex.

Reply:
Where would that input handler need to be added? Would it be another RequestInputType command handler that gets registered from the version specific code?

Reply:
It doesn't need to be a kernel command handler, but just something the kernel calls into when handling an input request. If you add custom handlers there for the hinted type, then use the custom experience. Otherwise, return a value that allows the central handler to fall back to the default behaviors (i.e. currently "file" and "text").

Reply:
Added a new handler